### ISSUE TITLE
Using Tailwind with new extensions, like MDX

### ISSUE THREAD

**Issue by samselikoff** [anantoghosh/gatsby-plugin-purgecss#408]
I added MDX and noticed some Tailwind classes were being incorrectly purged from my MDX files. Looking at the source I see that the Tailwind extractor only applies to these extensions:

https://github.com/anantoghosh/gatsby-plugin-purgecss/blob/c1be2496c02b3d651758fa1d7ce01f06b1ab8a1b/src/gatsby-node.js#L54

Would it be possible to generalize it a bit, ideally so it respects the `content` option? (I had already updated that to include `mdx` extension types).

**Comment by anantoghosh** [comment]
@samselikoff Oh I see. Try passing the another extractor with only the mdx extension in the plugin's `extractors` option.
If that doesn't work, then disable the `tailwind` option and pass the complete extractor in the `extractors` as you normally would.

**Comment by samselikoff** [comment]
> If that doesn't work, then disable the tailwind option and pass the complete extractor in the extractors as you normally would.

This is exactly what I ended up doing and it worked. Maybe we could provide the Tailwind extractor as an import?

```js
const tailwindExtractor = require("gatsby-plugin-purgecss/tailwind-extractor")

// ...

{
  resolve: `gatsby-plugin-purgecss`,
  options: {
    extractors: [
      {
        extractor: tailwindExtractor,
        extensions: ["js", "ts", "jsx", "tsx", "mdx"],
      },
    ],
    purgeOnly: ["/tailwind-utils.css"],
    content: [
      path.join(process.cwd(), "src/**/!(*.d).{ts,js,jsx,tsx,mdx}"),
    ],
  },
},
```

**Comment by anantoghosh** [comment]
I was thinking just adding the mdx in extensions array would be enough, since content option by default does not search for mdx, tailwind extractor will not do any extra work.
And in any other situation, the user is free to use the extractors as they want.

I may also include an option `mdx: true` for convenience.

**Comment by samselikoff** [comment]
Ah, just adding the extensions array makes it work!

```js
{
  resolve: `gatsby-plugin-purgecss`,
  options: {
    tailwind: true,
    extractors: [
      {
        extensions: ["js", "ts", "jsx", "tsx", "mdx"],
      },
    ],
    purgeOnly: ["/tailwind-utils.css"],
    content: [
      path.join(process.cwd(), "src/**/!(*.d).{ts,js,jsx,tsx,mdx}"),
    ],
  },
```

Imo `mdx: true` isn't necessary, what happens when another new extension comes out? No biggie just letting user configure their own.

**Comment by samselikoff** [comment]
I also think getting rid of `tailwind: true` option and letting folks import tailwind extractor would be an improvement. One less thing to learn.

**Comment by samselikoff** [comment]
This actually gives me an error:

> TypeError: Cannot read property 'extract' of undefined

**Comment by anantoghosh** [comment]
> Ah, just adding the extensions array makes it work!
Not what I meant. Adding the extension to this plugin's code should be enough, not passing it as an option.

> I also think getting rid of tailwind: true option and letting folks import tailwind extractor would be an improvement. One less thing to learn.
It's for convenience and is used a lot. I added it because people were using this plugin without using the tailwind extractor which caused problems.

I hope your issue is resolved 👍

**Comment by anantoghosh** [comment]
I've released v4.0.1 which adds md/mdx to the tailwind extractor. You don't need to specify the `extractors` option now.
The `content` option will still be needed.
